crates/ruff_linter/src/rules/pylint/rules/no_self_use.rs

```diff
@@ -119,3 +129,23 @@ pub(crate) fn no_self_use(checker: &Checker, scope: &Scope, diagnostics: &mut Ve
         ));
     }
 }
+
+struct CallVisitor {
+    is_super: bool,
+}
+
+impl<'a> Visitor<'a> for CallVisitor {
+    fn visit_expr(&mut self, expr: &'a Expr) {
+        match expr {
+            Expr::Call(ast::ExprCall { func, .. }) => match func.as_ref() {
+                Expr::Name(expr_name) => {
+                    self.is_super = expr_name.id == "super";
+                }
+                _ => {
+                    visitor::walk_expr(self, expr);
+                }
+            },
+            _ => visitor::walk_expr(self, expr),
+        }
+    }
+}
```

Comment:
Ideally, it would be better to use the function scope to check if the method contains a `super()` call in the body, but the function scope only gives me access to its parameters. So when using `scope.get("super")` it would only return `None`.

Reply:
It turns out that `super` is actually a global builtin in Python, so the usages we need are on the _global_ scope. I tweaked the logic to use the technique you suggested above. I guess the downside is that we can't ensure that the `super` usage was on a `super()` call, but perhaps that's fine. (We don't store a `NodeId` on `Reference`, though we could in the future.)